src/evaluators/BinaryExpression.js

```diff
@@ -170,8 +177,14 @@ export function computeBinary(
 
   if (lval instanceof AbstractValue || rval instanceof AbstractValue) {
     let type = getPureBinaryOperationResultType(realm, op, lval, rval, lloc, rloc);
-    return realm.createAbstract(new TypesDomain(type), ValuesDomain.topVal, [lval, rval], ([lnode, rnode]) =>
-      t.binaryExpression(op, lnode, rnode)
+    let [hash, args] = hashBinary(op, lval, rval);
+    return realm.createAbstract(
+      new TypesDomain(type),
+      ValuesDomain.topVal,
+      hash,
+      args,
+      ([lnode, rnode]) => t.binaryExpression(op, lnode, rnode),
+      op
```

Comment:
And this discussion is the reason why I'd like to move everything that is relevant to reasoning about abstract values in a separate ValueManager class. It should have a method like `createBinary(op, arg1, arg2)`, which hides the details and consolidates those implementation details.